pkg/storage/snapshot/restore_test.go

```diff
@@ -877,8 +881,24 @@ var _ = Describe("Restore controlleer", func() {
 						Expect(err).ShouldNot(HaveOccurred())
 					})
 
-				})
+					It("without instancetype and preference RevisionNames from original VM", func() {
+						// Add the revisionNames for the original VM instancetype and preference
+						r.Spec.Patches = []string{changeInstancetypePatch, changePreferencePatch}
+
+						vmInterface.EXPECT().Create(gomock.Any()).DoAndReturn(func(newVM *v1.VirtualMachine) (*v1.VirtualMachine, error) {
+							// Assert that these are then dropped when creating the new VirtualMachine
+							Expect(newVM.Spec.Instancetype.RevisionName).To(Equal(""))
+							Expect(newVM.Spec.Preference.RevisionName).To(Equal(""))
```

Comment:
nit: I think `BeEmpty()` is clearer

```go
Expect(newVM.Spec.Instancetype.RevisionName).To(BeEmpty())
Expect(newVM.Spec.Preference.RevisionName).To(BeEmpty())
```

wdyt? :)

Reply:
ACK makes sense thanks!